### PR TITLE
kubelet: update documentation of the cgroupDriver setting

### PR DESCRIPTION
**Pull request by marquiz**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Update the help text of the kubelet's `cgroupDriver` config setting (and the `--cgroup-driver` flag) to descrivbe 

#### Special notes for your reviewer:

Complements #118770

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pacoxu** [comment]
```
Please run 'hack/update-codegen.sh'
```
The change SGTM.

**Comment by bart0sh** [comment]
@marquiz please run hack/update-codegen.sh as suggested by @pacoxu above.
This should fix CI failure.

otherwise lgtm

**Comment by marquiz** [comment]
> please run hack/update-codegen.sh as suggested by

Thanks @bart0sh for pointing this out. Fixed.

I don't know how I've managed to turn down my own belowed PR

ping also @pacoxu

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -448,7 +448,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.StringVar(&c.ProviderID, "provider-id", c.ProviderID, "Unique identifier for identifying the node in a machine database, i.e cloudprovider")
 
 	fs.BoolVar(&c.CgroupsPerQOS, "cgroups-per-qos", c.CgroupsPerQOS, "Enable creation of QoS cgroup hierarchy, if true top level QoS and pod cgroups are created.")
-	fs.StringVar(&c.CgroupDriver, "cgroup-driver", c.CgroupDriver, "Driver that the kubelet uses to manipulate cgroups on the host.  Possible values: 'cgroupfs', 'systemd'")
+	fs.StringVar(&c.CgroupDriver, "cgroup-driver", c.CgroupDriver, "Driver that the kubelet uses to manipulate cgroups on the host.  Possible values: 'cgroupfs', 'systemd'. Note: ignored if the KubeletCgroupDriverFromCRI feature gate is enabled and a sufficiently new version of a container runtime is used, in which case the correct cgroup driver is automatically detected.")
```
Comment (bart0sh) [inline cmd/kubelet/app/options/options.go:451]: Can you be more specific about a container runtime version or re-frase this somehow? Do you mean the version that supports certain CRI API?

Comment (marquiz) [reply cmd/kubelet/app/options/options.go:451]: Yes, this is about a version of the container runtime that supports the `RuntimeConfig` CRI rpc. In practice this means CRI-O v1.28+ and containerd 2.0+ (2.0 is not released yet).

Sure, I can reword it. I feel that talking about RuntimeConfig rpc is not useful information to the end-user. Do we want to pintpoint specific versions of container runtimes here? WDYT?

Examples:

*Note: ignored if the KubeletCgroupDriverFromCRI feature gate is enabled and a container runtime supporting RuntimeConfig rpm is used, in which case the correct cgroup driver is automatically detected.")*

OR

*Note: ignored if the KubeletCgroupDriverFromCRI feature gate is enabled and a sufficiently new version of a container runtime is used (containerd v2.0 or later, cri-o v1.28 or later), in which case the correct cgroup driver is automatically detected.")*

Comment (bart0sh) [reply cmd/kubelet/app/options/options.go:451]: Thank you. I like the former better, except `rpm` thing. I don't understand it. Is it a typo and you meant `rpc`? 
Can we use `CRI API `or something similar instead?

Comment (marquiz) [reply cmd/kubelet/app/options/options.go]: Argh, my fat fingers and brain, rpm vs rpc 🙈

Changed to

*Note: ignored if the KubeletCgroupDriverFromCRI feature gate is enabled and a container runtime supporting the RuntimeConfig CRI rpc is used, in which case the correct cgroup driver is automatically detected.*

ping @sftim
you have educated opinions on these
/cc sftim

Comment (bart0sh) [reply cmd/kubelet/app/options/options.go]: Your last edition looks good to me. Please, update the PR and I'll lgtm it.

**Comment by bart0sh** [comment]
/assign @mrunalp @derekwaynecarr @dchen1107 
for a final approval

**Comment by dims** [comment]
@marquiz @SergeyKanzhelev i see the note to bump this feature to beta in https://github.com/kubernetes/enhancements/issues/4033#issuecomment-2127949657

is that in another PR or will this be updated?

**Pull request closed without merging**